packages/device-connector/src/arduino-cli/client.ts

```diff
@@ -272,38 +317,23 @@ export class RPCClient {
 
       const eventType = data.event_type
       const detectedPort = data.port
-      const port = detectedPort?.port
-      if (port == null) {
-        logger.error('Port not defined')
-        return
-      }
 
       if (eventType === 'add' && detectedPort?.matching_boards != null && detectedPort.matching_boards.length > 0) {
-        const board = detectedPort.matching_boards[0]
-        if (board.fqbn == null) {
-          logger.error('Could not register device: No fqbn found')
+        await this.addDevice(detectedPort)
+      } else if (eventType === 'remove') {
+        const port = detectedPort?.port
+        if (port == null) {
+          logger.warn('Port not defined')
           return
         }
 
-        const id = await registerNewDevice(board.fqbn, board.name ?? 'Unknown Device Name')
-
-        const device: Device = {
-          id,
-          name: board.name ?? 'Unknown Device Name',
-          fqbn: board.fqbn,
-          port: port
-        }
-
-        this.devices.push(device)
-        logger.info(`Device attached: ${device.name}`)
-      } else if (eventType === 'remove') {
-        if (port.address == null || port.address === '') {
+        if (port.address === undefined || port.address === '') {
```

Comment:
```suggestion
        if (port.address') {
```
assuming null is not supposed to be in the else branch

Reply:
Lint does not like this sadly.